aplt/metrics.py

```diff
@@ -0,0 +1,95 @@
+"""Metrics interface and implementations"""
+from twisted.internet import reactor
+from txstatsd.client import StatsDClientProtocol, TwistedStatsDClient
+from txstatsd.metrics.metrics import Metrics
+
+try:
+    import datadog
+    from datadog import ThreadStats
+    from datadog.util.hostname import get_hostname
+except ImportError:  # pragma: nocover
+    datadog = None
+
+
+class IMetrics(object):
+    """Metrics interface
+    Each method except :meth:`__init__` and :meth:`start` must be implemented.
+    Additional ``kwargs`` may be recorded as additional metric tags for metric
+    systems that support it, otherwise they should be ignored.
+    """
+    def __init__(self, *args, **kwargs):
+        """Setup the metrics"""
+
+    def start(self):
+        """Start any connection needed for metric transmission"""
+
+    def stop(self):
+        """Stop any connection needed"""
+
```

Comment:
Nope, nothing in a per-client scenario would make much sense to emit a gauge of.